src/core_modules/capture-core/components/Pages/MainPage/WorkingListsCommon/hooks/useWorkingListsCommonStateManagement.js

```diff
@@ -25,103 +25,151 @@ import {
+            workingListsTemplates[storeId].templates &&
+            workingListsTemplates[storeId].templates.find(template => template.id === workingListsTemplates[storeId].selectedTemplateId),
+        templates: workingListsTemplates[storeId] &&
+            workingListsTemplates[storeId].templates,
+        templatesLoading: !!workingListsTemplates[storeId] &&
+            !!workingListsTemplates[storeId].loading,
+        loadTemplatesError: workingListsTemplates[storeId] && workingListsTemplates[storeId].loadError,
     }), shallowEqual);
 
     const templateDispatch = useMemo(() => ({
-        onSelectTemplate: workingListDispatch(selectTemplate),
-        onLoadTemplates: workingListDispatch(fetchTemplates),
-        onCancelLoadTemplates: workingListDispatch(fetchTemplatesCancel),
-        onAddTemplate: workingListDispatch(addTemplate, (name: string, criteria: Object, data: Object) => [
-            name,
-            criteria, {
-                ...data,
-                listId,
-            },
-        ]),
-        onUpdateTemplate: workingListDispatch(updateTemplate, (template: Object, criteria: Object, data: Object) => [
-            template,
-            criteria, {
-                ...data,
-                listId,
-            },
-        ]),
-        onDeleteTemplate: workingListDispatch(deleteTemplate),
-        onCleanSkipInitAddingTemplate: workingListDispatch(cleanSkipInitAddingTemplate),
-    }), [listId, workingListDispatch]);
+        onSelectTemplate: (...args) => dispatch(selectTemplate(...args, storeId)),
+        onLoadTemplates: (...args) => dispatch(fetchTemplates(...args, storeId, workingListsType)),
+        onCancelLoadTemplates: (...args) => dispatch(fetchTemplatesCancel(...args, storeId)),
+        onAddTemplate: (name: string, criteria: Object, data: Object) =>
+            dispatch(addTemplate(
+                name,
+                criteria, {
+                    ...data,
+                    storeId,
+                    workingListsType,
+                },
+            )),
+        onUpdateTemplate: (template: Object, criteria: Object, data: Object) =>
+            dispatch(updateTemplate(
+                template,
+                criteria, {
+                    ...data,
+                    storeId,
+                    workingListsType,
+                },
+            )),
+        onDeleteTemplate: (...args) => dispatch(deleteTemplate(...args, { storeId, workingListsType })),
+        onCleanSkipInitAddingTemplate: (...args) => dispatch(cleanSkipInitAddingTemplate(...args, storeId)),
+    }: {|
+        onSelectTemplate: SelectTemplate,
+        onLoadTemplates: LoadTemplates,
+        onCancelLoadTemplates: CancelLoadTemplates,
+        onAddTemplate: AddTemplate,
+        onUpdateTemplate: UpdateTemplate,
+        onDeleteTemplate: DeleteTemplate,
+        onCleanSkipInitAddingTemplate: Function,
+    |}), [storeId, dispatch, workingListsType]);
```

Comment:
I am always thinking that `useCallback` is more preferred for memoizing functions. With `useMemo` basically we will have all the functions "calculated" again when `workingListsType` change but the `onCleanSkipInitAddingTemplate`, `onCancelLoadTemplates` and `onSelectTemplate` dont depend on it. 

We can both agree this wont be significant performance drawback however I am asking myself if using callback isnt a more graceful way to handle this?